samcli/lib/deploy/deployer.py

```diff
@@ -180,6 +179,8 @@ def create_changeset(
                 kwargs["TemplateURL"] = s3_uploader.to_path_style_s3_url(parts["Key"], parts.get("Version", None))
 
         # don't set these arguments if not specified to use existing values
+        if capabilities is not None:
```

Comment:
can you make it `if capabilities:`
I know this ways is more readable, but I am thinking if some how the passed capabilities is not None, but an empty list.

Reply:
This is to be consistent with the reset of arguments below, for example we consider an empty `notification_arns` array to be a valid value to pass  to cloudformation. I agree though that an empty array should be treated as none but I don't want to change the current behavior as it is a customer-facing change,

Reply:
ok, thanks for the clarification.